### PR TITLE
[Feature] Update assessment decisions

### LINKED ISSUE #8644: ✨ Update Assessment Decisions
## ✨ Feature

We need to update the assessment decisions that are possible with our application.  We are removing NOT_SURE and adding HOLD.

## 🕵️ Details

NOT_SURE is being removed in favor of using a null in the API to represent "no decision".

HOLD is being added as a sort of "soft successful".

## 🎨 Design File

@Jerryescandon 

Modals:
https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Pools---Record-of-Decision?type=design&node-id=6309-44439&mode=design&t=lVl0h2AgCTBVG24H-0

Step Tracker:
https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Pools---Record-of-Decision?type=design&node-id=6465-23869&mode=design&t=lVl0h2AgCTBVG24H-0

## 📸 Screenshot

image

image

## :gear: The assessment step algorithm

```
Essential requirements = essential skills AND education requirements
Optional requirements = asset skills
Assessment Step Status =
  IF any essential requirements unsuccessful THEN "unsuccessful"
  IF any requirements unassessed THEN "to assess"
  IF essential requirements include at least one hold THEN "hold"
  IF all essential requirements successful THEN "success"

Also, on the step tracker, a step effectively only has the "to assess" status if all previous steps are "hold" or "success".
```

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Remove the old decision and add the new decision to the AssessmentDecision PHP enum.
- [ ] Ensure that the assessment decision mutation can set the three values or a null.
- [ ] Update the modals created in #8397
  - [ ] The "Not sure" selection sets the decision back to null
  - [ ] The "Not demonstrated (Hold for further assessment)" selection sets the decision to HOLD.
- [ ] Update the step tracker created in #8530
  - [ ] Add the hold status to the summary
  - [ ] Update the logic as shown in the algorithm above and make sure it can be easily shared with the Assessment Step List

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8644 

## 👋 Introduction

Updates the assessment decisions adding an "on hold" and changing "not sure" to be `null`.

## 🕵️ Details

I was hoping to get something in the `App\Enums\AssessmentDecision` to be `NOT_SURE = null` but it requires a `string|int` and massaging the input seemed like too much trouble. Instead, a constant was added at `~/utils/assessmentResults` along with a modified `AssessmentDecision` type to allow it to be used in specific spots and then it is massaged on frontend. I'm not super happy with this approach, if anyone else has a better idea I'd be happy to try it out 😄 .

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run run dev`
2. Start storybook `npm run storybook:web`
3. Navigate to the "Assessment Step Tracker" story
4. Confirm the "on hold" status exists and is sorted properly (in between success and fail)
5. Confirm the counts are accurate
6. Navigate to the "ScreeningDecisionDialog" story
7. Confirm submitting with "Not sure" shows up as "noDecision" in the action data (this will be massaged to `null` at a higher level)
8. Confirm submitting with "Not demonstrated (Hold for further assessment)" shows up as "HOLD" on the action data

### Mutations

1. Navigate to `/graphqil`
2. Run the following mutations using `null` and `"HOLD"` as the `assessmentDecision` value
3. Confirm no errors

```graphql
mutation UpdateDecision($result: UpdateAssessmentResultInput!) {
  updateAssessmentResult(updateAssessmentResult: $result) {
    id
    assessmentDecision
  }
}

mutation CreateDescision($result: CreateAssessmentResultInput!) {
  createAssessmentResult(createAssessmentResult: $result) {
    id
    assessmentDecisionLevel
  }
}
```

## 📸 Screenshot

Screenshot 2024-01-04 143558
Screenshot 2024-01-04 143623

### CODE REVIEW

File: api/app/Enums/AssessmentDecision.php
Comment (petertgiles) [inline api/app/Enums/AssessmentDecision.php:7]: Could you update api/database/factories/AssessmentResultFactory.php to sometimes seed nulls?

**Review by petertgiles — CHANGES_REQUESTED** [review]
I did a quick readthrough and it looks pretty good.  It's not building right now though.  Could you take a look?

File: api/app/Enums/AssessmentDecision.php
Comment (esizer) [reply api/app/Enums/AssessmentDecision.php:7]: Good call! Added in [add null decision to factory](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8955/commits/0d94c66619381d7cedd5a6c9618fb47ebbe2e12a)

Also, I didn't realize we added a test that used the old value, should build now after [fix tracker tests after merge](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8955/commits/7e962fe9ed931ffe1657ebf808dab461b62735b1)

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx
Comment (petertgiles) [inline apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx:57]: I don't think you can double these up.  For WATCH you need to reset justifications, I think.
image

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looks great! :medal_sports:  I think just one last thing to check on.

File: apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx
Comment (esizer) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx]: Okay, I was a little confused by this specific part and I was looking into this and think I messed it up. It seems like maybe this auto sets it to "other" and then shows the notes? 🤔 

Screenshot 2024-01-08 153009

Comment (petertgiles) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx]: Oh yes, you're right!  There should be the copy block and "Other reason" input.

Comment (esizer) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx]: Alright, I _think_ I got it in [update justification logic for on hold decision](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8955/commits/27e492e2c0e33e11cc1abbfada27ae3012023a4e)

Comment (petertgiles) [reply apps/web/src/components/ScreeningDecisions/ScreeningDecisionDialogForm.tsx]: Hmm, that's an interesting way to do it - set the other justification in a hidden list.  I don't mind, I guess it makes sense.

**Review by petertgiles — APPROVED** [review]

**Pull request merged**
